### PR TITLE
Learnable temperature refactor

### PR DESCRIPTION
**Pull request by Simontwice**
# Description
needed for training temperature post main training loop

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: research/conditional/utils/model_utils.py [hunk 1]
```diff
@@ -387,6 +390,20 @@ def get_ff_layer(args):
             share_by_experts=args.share_by_experts,
             share_by_emit_merge=args.share_by_emit_merge,
         )
+    elif args.ff_mode == "cont_moe_adatemp":
```
Comment (Simontwice) [inline research/conditional/utils/model_utils.py:393]:
```suggestion
    elif args.ff_mode == "cont_moe_adatemp_positive":
```

File: research/conditional/moe_layers/continuous_moe.py [hunk 2]
```diff
@@ -46,17 +46,17 @@ def __post_init__(self):
                 f"expert_size is None, setting it to dff // n_experts = {self.dff // self.n_experts}"
             )
             self.expert_size = self.dff // self.n_experts
-        self.init_parameters()
+        self.init_core_parameters()
+        self.init_additional_parameters()
```
Comment (crewtool) [inline research/conditional/moe_layers/continuous_moe.py:50]: Why such distinction?

File: research/conditional/moe_layers/continuous_moe.py [hunk 3]
```diff
@@ -46,17 +46,17 @@ def __post_init__(self):
...
         self.original_group_size = self.group_size
 
     def forward(self, x):
         x = self.rearrange_for_grouping(x)
+        merge_weights, emit_weights = self.get_merge_and_emit_weights(x)
+        print("merge_weights", merge_weights.shape)
```
Comment (crewtool) [inline research/conditional/moe_layers/continuous_moe.py:56]: Remove print

**Review by crewtool — APPROVED** [review]
LGTM
remove debug print

File: research/conditional/moe_layers/continuous_moe.py (hunk 2 above)
Comment (Simontwice) [reply research/conditional/moe_layers/continuous_moe.py:50]: It allows child layers to introduce new parameters, without copypasting init of the existing parameters.

File: research/conditional/moe_layers/continuous_moe.py (hunk 3 above)
Comment (Simontwice) [inline research/conditional/moe_layers/continuous_moe.py:56]:
```suggestion
```

Comment (Simontwice) [reply research/conditional/moe_layers/continuous_moe.py]: done!

**Pull request merged**